zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -873,6 +873,10 @@ public int getGlobalOutstandingLimit() {
         return limit;
     }
 
+    public static long getSnapSize() {
+        return Long.getLong("zookeeper.snapSize", 4294967296L); // 4GB by default
```

Comment:
We should document this (probably in zookeeperAdmin.md).

Maybe by default should be off with a special value like -1L? For backward compatibility reasons, but it doesn't really break anything... maybe just wait for a binding vote on this :)

Reply:
Excellent point! I've extended the documentation to describe the new property. Let's leave it on by default and let the community decide if they want to modify that value.